x/gamm/client/cli/tx.go

```diff
@@ -178,6 +181,115 @@ func NewStableSwapAdjustScalingFactorsCmd() *cobra.Command {
 	return cmd
 }
 
+// NewCmdSubmitReplaceMigrationRecordsProposal implements a command handler for replace migration records proposal
+func NewCmdSubmitReplaceMigrationRecordsProposal() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "replace-migration-records-proposal [flags]",
+		Args:  cobra.ExactArgs(0),
+		Short: "Submit a replace migration record proposal",
+		Long: strings.TrimSpace(`Submit a replace migration record proposal.
+
+Passing in poolIds separated by commas would be parsed automatically to pairs of migration record.
+Ex) 2,4,1,5 -> [(Balancer 2, CL 4), (Balancer 1, CL 5)]
+
+
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			content, err := parseReplaceMigrationRecordsArgsToContent(cmd)
+			if err != nil {
+				return err
+			}
+
+			from := clientCtx.GetFromAddress()
+
+			depositStr, err := cmd.Flags().GetString(govcli.FlagDeposit)
+			if err != nil {
+				return err
+			}
+			deposit, err := sdk.ParseCoinsNormalized(depositStr)
+			if err != nil {
+				return err
+			}
+
+			msg, err := govtypes.NewMsgSubmitProposal(content, deposit, from)
+			if err != nil {
+				return err
+			}
+
+			if err = msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	cmd.Flags().String(govcli.FlagTitle, "", "title of proposal")
+	cmd.Flags().String(govcli.FlagDescription, "", "description of proposal")
+	cmd.Flags().String(govcli.FlagDeposit, "", "deposit of proposal")
+	cmd.Flags().String(FlagMigrationRecords, "", "The migration records array")
+
+	return cmd
+}
+
+// NewCmdSubmitUpdateMigrationRecordsProposal implements a command handler for replace migration records proposal
```

Comment:
```suggestion
// NewCmdSubmitUpdateMigrationRecordsProposal implements a command handler for update migration records proposal
```

Reply:
@czarcas7ic thanks for flagging! Here is the issue that I have created to track this https://github.com/osmosis-labs/osmosis/issues/4672